enumerables.rb

```diff
@@ -0,0 +1,67 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each { |i| array << i if yield(i) }
+    array
+  end
+
+  def my_all?
+    output = true
+    my_each { |i| break output = false unless yield(i) }
+    output
+  end
```

Comment:
- returns false if the block returns false or nil
- when no block or argument is given returns true when none of the collection members are false or nil
- when no block or argument is given returns false when one of the collection members are false or nil
- when a class is passed as an argument returns true if all of the collection is a member of such class
- when a Regex is passed as an argument returns true if all of the collection matches the Regex
- when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern